OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameChatLogic.cs

```diff
@@ -60,11 +64,15 @@ public IngameChatLogic(Widget widget, OrderManager orderManager, World world, Mo
 			tabCompletion.Commands = chatTraits.OfType<ChatCommands>().SelectMany(x => x.Commands.Keys).ToList();
 			tabCompletion.Names = orderManager.LobbyInfo.Clients.Select(c => c.Name).Distinct().ToList();
 
+			if (logicArgs.TryGetValue("Templates", out var templateIds))
+				foreach (var item in templateIds.ToDictionary(kv => kv.Value))
+					templates[item.Key] = Ui.LoadWidget(item.Value, null, new WidgetArgs());
```

Comment:
How about something like this?
```csharp
if (logicArgs.TryGetValue("Templates", out var templateIds))
{
	foreach (var item in templateIds.Nodes)
	{
		var key = FieldLoader.GetValue<TextNotificationPool>("key", item.Key);
		templates[key] = Ui.LoadWidget(item.Value.Value, null, new WidgetArgs());
	}
}
```

This would avoid the `ToDictionary` call, and allows that `templates` to key on `TextNotificationPool` to save the `ToString`s when adding lines and better match `TextNotificationsDisplayWidget`.

(and likewise in `LobbyLogic`).